Runtime/Scripts/DeviceInfoIntegration.cs

```diff
@@ -0,0 +1,155 @@
+        return null;
+      }
+      map = new Dictionary<string, string>();
+
+      AndroidJavaClass versionClass = new AndroidJavaClass("android.os.Build$VERSION");
+      int sdk_int = PlatformIntegrationUtil.GetStatic<int>(versionClass, "SDK_INT");
+      map["Build.VERSION.SDK_INT"] = sdk_int.ToString();
+
+      const string readPhoneStatePermissionString = "READ_PHONE_STATE";
+      try
+      {
+        if (Permission.HasUserAuthorizedPermission(readPhoneStatePermissionString))
+        {
+          string ver = PlatformIntegrationUtil.Call<string>(telephonyManager, "getDeviceSoftwareVersion");
+          if (ver != null)
+          {
+            map["DeviceSoftwareVersion"] = ver.ToString();
+          }
+        }
+      }
+      catch (Exception e)
+      {
+        Debug.Log("Exception retrieving properties: " + e.GetBaseException() + ", " + e.Message);
+      }
+
+      try
+      {
+        if (Permission.HasUserAuthorizedPermission(readPhoneStatePermissionString))
+        {
+          int nType = PlatformIntegrationUtil.Call<int>(telephonyManager, "getDataNetworkType");
+          NetworkDataType datatype = (NetworkDataType)nType;
+          map["DataNetworkType"] = datatype.ToString();
+        }
+      }
+      catch (Exception e)
+      {
+        Debug.Log("Exception retrieving properties: " + e.GetBaseException() + ", " + e.Message);
+      }
+
+      AndroidJavaClass versionCodesClass = new AndroidJavaClass("android.os.Build$VERSION_CODES");
+      int versionCode = PlatformIntegrationUtil.GetStatic<int>(versionCodesClass, "Q");
+      if (sdk_int > versionCode)
+      {
+        string mc = PlatformIntegrationUtil.Call<string>(telephonyManager, "getManufacturerCode");
+        if (mc != null)
+        {
+          map["ManufacturerCode"] = mc;
+        }
+      }
+
+      string niso = PlatformIntegrationUtil.Call<string>(telephonyManager, "getNetworkCountryIso");
+      if (niso != null)
+      {
+        map["NetworkCountryIso"] = niso;
+      }
+
+      string siso = PlatformIntegrationUtil.Call<string>(telephonyManager, "getSimCountryIso");
+      if (siso != null)
+      {
+        map["SimCountryCodeIso"] = siso;
+      }
+
+      int phoneType = PlatformIntegrationUtil.Call<int>(telephonyManager, "getPhoneType");
+      map["PhoneType"] = phoneType.ToString();
+
+      // Default one.
+      string simOperatorName = PlatformIntegrationUtil.Call<string>(telephonyManager, "getSimOperatorName");
+      if (simOperatorName != null)
+      {
+        map["SimOperatorName"] = simOperatorName;
+      }
+
+      // Default one.
+      string networkOperator = PlatformIntegrationUtil.Call<string>(telephonyManager, "getNetworkOperatorName");
+      if (networkOperator != null)
+      {
+        map["NetworkOperatorName"] = networkOperator;
+      }
+
+      return map;
+    }
+#elif UNITY_IOS
+  public Dictionary<string, string> GetDeviceInfo()
+  {
+    Debug.LogFormat("[{0}] DeviceInfo not implemented!", TAG);
```

Comment:
Do you have this TAG defined somewhere?

Reply:
Nowhere. You'll hit a compile error right away, asking you to define it or stub it. We do plan to implement IOS, so we can revisit again. Or add TAG = "name of class"